src/vm/amd64/pinvokestubs.S

```diff
@@ -130,24 +130,90 @@ NESTED_END VarargPInvokeGenILStub, _TEXT
 
 //
 // IN:
-// InlinedCallFrame (rcx) = pointer to the InlinedCallFrame data, including the GS cookie slot (GS cookie right 
+// InlinedCallFrame (rdi) = pointer to the InlinedCallFrame data, including the GS cookie slot (GS cookie right 
 //                          before actual InlinedCallFrame data)
 //
 //
-LEAF_ENTRY JIT_PInvokeBegin, _TEXT
-        // Not yet supported
-        int 3
+NESTED_ENTRY JIT_PInvokeBegin, _TEXT, NoHandler
+
+        push_nonvol_reg r14
+        END_PROLOGUE
+
+        PREPARE_EXTERNAL_VAR s_gsCookie, rax
+        mov             rax, qword ptr [rax]
+        mov             qword ptr [rdi], rax
+        add             rdi, SIZEOF_GSCookie
+
+        // set first slot to the value of InlinedCallFrame vftable (checked by runtime code)
+#ifdef __APPLE__
+        PREPARE_EXTERNAL_VAR C_FUNC(ZTV16InlinedCallFrame), rax
```

Comment:
It looks like the unification would be:
`PREPARE_EXTERNAL_VAR _ZTV16InlinedCallFrame, rax`
C_FUNC prepends underscore on Apple and does nothing on other systems.

Reply:
Hmm... I'm thought that's what I tried in the previous iteration and the CI failed on Linux. I'll take another look, maybe i missed something before

Reply:
Ok, it worked without the C_FUNC this time. Fixed.